vm/class_test.go

```diff
@@ -990,3 +990,40 @@ func TestClassSingletonClassClassMethod(t *testing.T) {
 		v.checkSP(t, i, 1)
 	}
 }
+
+func TestObjectId(t *testing.T) {
+	tests := []struct {
+		input    string
+		expected interface{}
+	}{
+		// immutable objects
+		{`Object.new.object_id.is_a?(Integer)`, true},
+		{`a = nil.object_id; b = nil.object_id; a == b`, true},
+		{`a = true.object_id; b = true.object_id; a == b`, true},
+		{`a = false.object_id; b = false.object_id; a == b`, true},
+		{`CONSTANT=1; CONSTANT.object_id == CONSTANT.object_id`, true},
+		{`Class.object_id == Class.object_id`, true},
+		{`Object.object_id == Object.object_id`, true},
+		{`Integer.object_id == Integer.object_id`, true},
+		// mutable objects
+		{`a = 1.object_id; b = 1.object_id; a == b`, false},
```

Comment:
I don't think this means objects are mutable, we just create a new one every time.

Reply:
I'll fix this.